Geometry/HGCalCommonData/src/HGCalGeomParameters.cc

```diff
@@ -891,23 +894,48 @@ void HGCalGeomParameters::loadWaferHexagon(HGCalParameters& php) {
 	  php.waferPosY_.emplace_back(ypos);
 	  for (unsigned int il=0; il<php.layer_.size(); ++il) {
 	    bool corner(false), cornerAll(true);
+	    int  nCorner(0), firstCorner(-1), firstMiss(-1);
 	    for (unsigned int k=0; k<HGCalParameters::k_CornerSize; ++k) {
 	      double rpos = std::sqrt(xc[k]*xc[k]+yc[k]*yc[k]);
 	      if (rpos >= php.rMinLayHex_[il] && 
-		  rpos <= php.rMaxLayHex_[il]) corner    = true;
-	      else                             cornerAll = false;
+		  rpos <= php.rMaxLayHex_[il]) {
+		corner    = true;
+		++nCorner;
+		if (firstCorner < 0) firstCorner = k;
+	      } else {
+		cornerAll = false;
+		if (firstMiss < 0)   firstMiss = k;
+	      }
 	    }
 	    if (corner) {
 	      auto cpy = copiesInLayers[php.layer_[il]].find(copy);
 	      if (cpy == copiesInLayers[php.layer_[il]].end()) 
 		copiesInLayers[php.layer_[il]][copy] = cornerAll ? php.waferCopy_.size() : -1;
 	    }
+	    if ((!cornerAll) && (nCorner > 0)) {
+	      if (nCorner == 3) {
+		if ((firstCorner == 0) && (firstMiss < 3)) 
```

Comment:
it seems like these nested if conditions could be combined

Reply:
Done